db/agendadb/db.go

```diff
@@ -0,0 +1,217 @@
+// and a string array suitable for use as a primary key. The AgendaID is tagged
+// as an index for quick lookups based on the agenda.
+type ChoiceLabeled struct {
+	AgendaChoice   [2]string `storm:"id"`
+	AgendaID       string    `json:"agendaid" storm:"index"`
+	dcrjson.Choice `storm:"inline"`
+}
+
+// GetVoteAgendasForVersion is used in getting the agendas using the vote versions
+func GetVoteAgendasForVersion(ver int64, client *rpcclient.Client) (agendas []AgendaTagged) {
+	voteInfo, err := client.GetVoteInfo(uint32(ver))
+	if err == nil {
+		for _, v := range voteInfo.Agendas {
+			a := AgendaTagged{
+				Id:             v.Id,
+				Description:    v.Description,
+				Mask:           v.Mask,
+				StartTime:      v.StartTime,
+				ExpireTime:     v.ExpireTime,
+				Status:         v.Status,
+				QuorumProgress: v.QuorumProgress,
+				Choices:        v.Choices,
+				VoteVersion:    voteInfo.VoteVersion,
+			}
+			agendas = append(agendas, a)
+		}
+	}
+	return
+}
+
+// CheckAvailabiltyOfVersionAgendas checks for the availabily of agendas in the
+// saved db by version input
+func (db *AgendaDB) CheckAvailabiltyOfVersionAgendas(version int64) bool {
+	if db == nil || db.sdb == nil {
+		return false
+	}
+	agenda := new([]AgendaTagged)
+	if err := db.sdb.Find("VoteVersion", uint32(version), agenda); err != nil {
+		return false
+	}
+	if agenda == nil {
+		return false
+	}
+	return true
+}
+
+// Updatedb used when needed to keep the saved db upto date
+func (db *AgendaDB) Updatedb(voteVersion int64, client *rpcclient.Client) {
+	var agendas []AgendaTagged
+	for GetVoteAgendasForVersion(voteVersion, client) != nil {
+		agendas = append(agendas, GetVoteAgendasForVersion(voteVersion, client)...)
+		voteVersion++
+	}
+	for _, a := range agendas {
+		err := db.StoreAgenda(&a)
+		if err != nil {
+			log.Errorf("Agenda not saved: %v \n", err)
+		}
+	}
+}
+
+// CheckForUpdates checks for update at the start of the process and
+// will proceed to update when neccessary
+func CheckForUpdates(client *rpcclient.Client) {
+	adb, err := Open(dbName)
+	if err != nil {
+		log.Errorf("Failed to open new DB: %v", err)
+	}
+	var voteVersion int64 = 4
+	for adb.CheckAvailabiltyOfVersionAgendas(voteVersion) {
+		voteVersion++
+	}
+	adb.Updatedb(voteVersion, client)
+	adb.Close()
+}
+
+// GetAgendaInfo for getting an agenda's details given it's agendaId
+func GetAgendaInfo(agendaId string) (*AgendaTagged, error) {
+	adb, err := Open(dbName)
+	if err != nil {
+		log.Errorf("Failed to open new DB: %v", err)
+	}
+	agenda, err := adb.LoadAgenda(agendaId)
+	if err != nil {
+		log.Error(err)
+		return nil, err
+	}
+	adb.Close()
+	return agenda, nil
+}
+
+// GetAllAgendas returns all agendas and their info in the db
+func GetAllAgendas() (agendas []*AgendaTagged) {
+	adb, err := Open(dbName)
+	if err != nil {
+		log.Errorf("Failed to open new DB: %v", err)
+	}
+	err = adb.sdb.All(&agendas)
+	if err != nil {
+		log.Errorf("Failed to get data from DB: %v", err)
```

Comment:
Most likely these errors should be handled higher in the call stack.

Reply:
Some errors may still be passed from the higher call stack especially if the storm db initialization had errors.